flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/StopWithSavepointOperations.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.scheduler;
+
+import org.apache.flink.runtime.execution.ExecutionState;
+import org.apache.flink.runtime.executiongraph.Execution;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * {@code StopWithSavepointOperations} collects the steps for creating a savepoint and waiting for
+ * the job to stop.
+ */
+public interface StopWithSavepointOperations {
+
+    /**
+     * Handles the Savepoint creation termination.
+     *
+     * @param path the path to the newly created savepoint.
+     * @param throwable the {@code Throwable} in case of failure.
```

Comment:
What is the contract between `path` and `throwable`?

Reply:
Maybe it is a bit easier to have `onSavepointSuccess(String path)` and `onSavepointFailure(Throwable throwable)`

Reply:
I followed your advice and added to methods for the two cases. That helps differentiating the two cases! 👍